src/components/ScholarshipDetailCard.jsx

```diff
@@ -111,7 +112,16 @@ export default function ScholarshipDetailCard({ scholarship }) {
         <Button
           variant="outlined"
           className={classes.shareBtn}
-          startIcon={<Share />}>
+          startIcon={<Share />}
+          onClick={() =>
+            history.replace({
+              state: {
+                showShareDialog: true,
+                shareTitle: `${amount?.toString()} - ${name}`,
+                shareURL: `https://${window.location.hostname}/scholarships/${scholarship.id}`,
+              },
+            })
+          }>
```

Comment:
What's the issue with doing it this way?
```js
import { Link } from 'react-router-dom';
```
```suggestion
                component={Link}
                to={{ state: {
                  showShareDialog: true,
                  shareTitle: `${amount?.toString()} - ${name}`,
                  shareURL: `https://${window.location.hostname}/scholarships/${scholarship.id}`,
                } }}
                replace>
```

Reply:
The problem was that I was using the wrong `Link`